src/greenlight/runner.clj

```diff
@@ -77,6 +102,68 @@
     true))
 
 
+(defmacro ^:private with-delayed-output
+  [printer & body]
+  `(let [out# (java.io.StringWriter.)
+         original-out# *out*
+         original-err# *err*
+         printer# ~printer]
+     (binding [*out* out#, *err* out#]
+       (let [result# (do ~@body)]
+         (binding [*out* original-out#
+                   *err* original-err#]
+           (printer# (str out#))
+           result#)))))
```

Comment:
So stderr is effectively getting directed to stdout? That seems potentially expectation-breaking, but on the other hand, keeping two StringWriters and printing them both out at the end would result in reordered log lines, which could be even worse.

Reply:
Yeah, I thought leaving the ordering the same was the lesser of two evils here.

Reply:
There is a much more complicated thing we could do here to preserve 'stderr-ness', but I think this is fine for now.